controllers/profiles/reconciler_dev.go

```diff
@@ -120,9 +127,30 @@ func (e *ensureRunningDevWorkflowReconciliationState) Do(ctx context.Context, wo
 	}
 	objs = append(objs, propsCM)
 
+	devBaseContainerImage := defaultKogitoServerlessWorkflowDevImage + ":" + version.OperatorVersion
+	pl, errPl := platform.GetActivePlatform(ctx, e.client, workflow.Namespace)
+	// check if the Platform available
+	if errPl == nil {
+
+		// if is present the optional devBaseImage
+		if len(pl.Spec.DevBaseImage) > 0 {
+			devBaseContainerImage = pl.Spec.DevBaseImage
+		} else {
+			// If is not present the optional devBaseImage and is present the baseImage
+			if len(pl.Spec.BuildPlatform.BaseImage) > 0 {
+				devBaseContainerImage = pl.Spec.BuildPlatform.BaseImage
+			}
+		}
+	} else {
+		// is the operator version is a snapshot we add the nightly version
+		if isSnapshot(version.OperatorVersion) {
+			devBaseContainerImage = defaultKogitoServerlessWorkflowDevImage + "-" + nightlySuffix + ":latest"
+		}
+	}
+
 	deployment, _, err := e.ensurers.deployment.ensure(ctx, workflow,
 		defaultDeploymentMutateVisitor(workflow),
-		naiveApplyImageDeploymentMutateVisitor(defaultKogitoServerlessWorkflowDevImage),
+		naiveApplyImageDeploymentMutateVisitor(devBaseContainerImage),
```

Comment:
Isn't it better to pass the dev image to use to the `naiveApplyImageDeploymentMutateVisitor`